lib/memo_wise.rb

```diff
@@ -94,42 +183,164 @@ def #{method_name}#{args_str}
     end
   end
 
-  def reset_memo_wise(method_name, *args, **kwargs)
+  # Presets the memoized result for the given method to the result of the given
+  # block.
+  #
+  # This method is for situations where the caller *already* has the result of
+  # an expensive method call, and wants to preset that result as memoized for
+  # future calls. In other words, the memoized method will be called *zero*
+  # times rather than once.
+  #
+  # @param method_name [Symbol]
+  #   Name of a method previously setup with `#memo_wise`.
+  #
+  # @param args [Array]
+  #   (Optional) If the method takes positional args, these are the values of
+  #   position args for which the given block's result will be preset as the
+  #   memoized result.
+  #
+  # @param kwargs [Hash]
+  #   (Optional) If the method takes keyword args, these are the keys and values
+  #   of keyword args for which the given block's result will be preset as the
+  #   memoized result.
+  #
+  # @yieldreturn [Object]
+  #   The result of the given block will be preset as memoized for future calls
+  #   to the given method.
+  #
+  # @return [void]
+  #
+  # @example
+  #   class Example
+  #     prepend MemoWise
+  #     attr_reader :method_called_times
+  #
+  #     def method_to_preset
+  #       @method_called_times = (@method_called_times || 0) + 1
+  #       "A"
+  #     end
+  #     memo_wise :method_to_preset
+  #   end
+  #
+  #   ex = Example.new
+  #
+  #   ex.preset_memo_wise(:method_to_preset) { "B" }
+  #
+  #   ex.method_to_preset #=> "B"
+  #
+  #   ex.method_called_times #=> nil
+  #
+  def preset_memo_wise(method_name, *args, **kwargs)
     validate_memo_wised!(method_name)
 
-    unless method_name.is_a?(Symbol)
-      raise ArgumentError, "#{method_name.inspect} must be a Symbol"
+    unless block_given?
```

Comment:
Just want to make sure I'm reading this diff right -- it's appearing like this because of the reordering of the methods right, not because there are actual changes to `preset_memo_wise`?

Reply:
Hi @jemmaissroff can you perhaps review the proposed new version of `preset_memo_wise` (that is, rather than the diff with the old version) and see if that answers this question?

Reply:
@jemmaissroff is this view helpful, to see the final version of `preset_memo_wise` rather than the diff?

https://github.com/panorama-ed/memo_wise/blob/7aa734d97948cc71330d66fb3c5694f222317d13/lib/memo_wise.rb#L233

Reply:
Looks great, thanks for copying it in here